packages/runtime/src/lib/asa.ts

```diff
@@ -1,36 +1,39 @@
 import path from "path";
 import * as z from 'zod';
 
-import { BuilderError } from "../internal/core/errors";
-import { ERRORS } from "../internal/core/errors-list";
-import { ASSETS_DIR } from "../internal/core/project-structure";
-import { parseZodError } from "../internal/core/validation-errors";
-import { AccountMap, ASADef, ASADefs } from "../types";
+import { BuilderError } from "../errors/errors";
+import { ERRORS } from "../errors/errors-list";
+import { parseZodError } from "../errors/validation-errors";
+import { AccountMap, ASADef, ASADefs, RuntimeAccountMap } from "../types";
 import { ASADefsSchema } from "../types-input";
 import { loadFromYamlFileSilentWithMessage } from "./files";
 
-function validateSingle (accounts: AccountMap, filename: string, asaDef: ASADef): void {
+const ASSETS_DIR = "assets";
+
+function validateSingle (accounts: AccountMap | RuntimeAccountMap, filename: string, asaDef: ASADef): void {
   if (!asaDef.optInAccNames || asaDef.optInAccNames.length === 0) {
     return;
   }
   for (const accName of asaDef.optInAccNames) {
     if (!accounts.get(accName)) {
       throw new BuilderError(
-        ERRORS.SCRIPT.ASA_PARAM_ERROR_NO_NAMED_OPT_IN_ACCOUNT, {
+        ERRORS.ASA.ASA_PARAM_ERROR_NO_NAMED_OPT_IN_ACCOUNT, {
           filename: filename,
           optInAccName: accName
         });
     }
   }
 }
 
-function validateParsedASADefs (accounts: AccountMap, asaDefs: ASADefs, filename: string): void {
+function validateParsedASADefs (
+  accounts: AccountMap | RuntimeAccountMap, asaDefs: ASADefs, filename: string): void {
```

Comment:
Please add a doc string explaining the two account types. `AccountMap` is the SDK account type, used in builder. `RuntimeAccountMap` is for `StoreAccount` used in runtime (where we use maps instead of arrays in sdk structures).

Reply:
yes